src/simulator/shape.py

```diff
@@ -13,3 +14,21 @@ def bounding_box(self, position: Position):
 
     def dump_info(self):
         pass
+
+    def rotation(x: float, y: float, z: float, phi: float, theta: float) -> list[float]:
+        """rotate the point by angles
+        Args:
+            x,y,z : coordinate of point in Cartesian
+            phi : angle of rotation about z-axes respectively
+            theta : angle of rotation about y-axes respectively
+        Return:
+            list of new coordinate point after rotation
+        """
+        cos_theta = cos(deg2rad(theta))
+        cos_phi = cos(deg2rad(phi))
+        sin_theta = sin(deg2rad(theta))
+        sin_phi = sin(deg2rad(phi))
+        new_x = round(x * cos_phi*cos_theta - y * sin_phi + z * sin_theta * cos_phi, 3)
```

Comment:
Please implement the rotation using matrices. I agree that the principle is the same but numpy matrix multiplication might be very optimized. And the math would be much more clear.